frame/staking/src/lib.rs

```diff
@@ -427,6 +427,21 @@ pub enum StakerStatus<AccountId> {
 	Nominator(Vec<AccountId>),
 }
 
+/// Policy for receiving rewards
+#[derive(PartialEq, Eq, Copy, Clone, Encode, Decode, RuntimeDebug)]
+pub enum RewardPolicy<AccountId> {
```

Comment:
This is not a backwards compatible decision... I think you should simply extend `RewardDestination` if possible.

Reply:
Good point, was considering adding another variant to `RewardDestination`
```rust
pub enum RewardDestination<AccountId> {
   ...
   Split(AccountId,Perbill,AccountId),
}
```
The problem is like you said if both accounts are dead accounts. Since, we know the `RewardDestination::Stash` and `RewardDestination::Staked` are not dead, does that mean we could instead just do something like
```rust
pub enum RewardDestination<AccountId> {
   ...
   Split1(AccountId,Perbill,Staked),
   Split2(AccountId,Perbill,Stash),
   Split3(Stash,Perbill,Staked),
}
```
and it wouldn't change the worst case?

Reply:
In either case it would require migration and is a breaking change for clients. I think a composable design (a _policy_ that contains combination of two _destinations_) is sensible, regardless of the naming.

Reply:
@kianenigma how would extending reward destination be a breaking change?

Reply:
Yes I was simply wrong :D

Reply:
Although, in my initial thought `Split(AccountId, Perbill, AccountId)` is not sensible because the former `AccountId` is assumed to be `Staked`, and there's no clear way to express 10% random account (e.g. my accountant) and rest my controller.